onnxscript/rewriter/pattern.py

```diff
@@ -1812,10 +1825,20 @@ def print(self):
                     print(f"Graph matching failed: {reason}")
             else:
                 print("Graph matching failed.")
-            failure_node = self.match_result._failure_node
-            if failure_node:
-                print("Failure at or around node:")
-                failure_node.display()
+            failure_nodes_and_values = self.match_result._failure_nodes_and_values
+            print("Failure at or around nodes/values:")
+
+            if failure_nodes_and_values:
+                if isinstance(failure_nodes_and_values, list):
+                    for failure_cause in failure_nodes_and_values:
+                        if isinstance(failure_cause, ir.Node):
+                            failure_cause.display()
+                        elif isinstance(failure_cause, ir.Value):
+                            print(failure_cause)
+                elif isinstance(failure_nodes_and_values, ir.Node):
+                    failure_nodes_and_values.display()
+                elif isinstance(failure_nodes_and_values, ir.Value):
+                    print(failure_nodes_and_values)
```

Comment:
Even a value has a display method, so you can call it, I think. (But there may be scope for improving the display of values ... but that is an orthogonal issue, can deal with it separately.)

Reply:
The display methods were designed for terminal interactive use. Would print be enough? If not we can improve

Reply:
Using the display call for now, we can improve if needed.